src/renderer/ThemeBackend.js

```diff
@@ -0,0 +1,225 @@
+      true, _ => Color('#4c6e7d').lighten(0.24).hex()
+    ),
+    chatListItemSelectedText: theme.bgPrimary,
+    chatListItemBgHover: undefinedGuard(
+      theme.bgPrimary, c => changeContrast(c, 0.3)
+    ),
+    // Message Bubble
+    messageText: theme.textPrimary,
+    messageTextLink: theme.textPrimary, // same as message text
+    setupMessageText: '#ed824e',
+    infoMessageBubbleBg: '#000000',
+    infoMessageBubbleText: 'white',
+    messageIncommingBg: '#ffffff',
+    messageIncommingDate: theme.textPrimary,
+    messageOutgoingBg: '#efffde',
+    messageOutgoingStatusColor: '#4caf50',
+    // Message Bubble - Buttons
+    messageButtons: '#8b8e91',
+    messageButtonsHover: '#070c14',
+    // Message Bubble - Metadata
+    messageStatusIcon: '#a4a6a9',
+    messageStatusIconSending: '#62656a',
+    messagePadlockOutgoing: '#4caf50',
+    messagePadlockIncomming: '#a4a6a9',
+    messageMetadataImageNoCaption: '#ffffff',
+    messageMetadataDate: '#62656a',
+    messageMetadataIncomming: 'rgba(#ffffff, 0.7)',
+    messageAuthor: '#ffffff',
+    // Message Bubble - Attachments
+    messageAttachmentIconExtentionColor: '#070c14', // Only changable with theme.raw
+    messageAttachmentIconBg: '#ffffff', // Only changable with theme.raw
+    messageAttachmentFileName: '#070c14',
+    messageAttachmentFileSize: '#070c14',
+    // Login Screen
+    loginInputFocusColor: '#42A5F5',
+    loginButtonText: '#42A5F5',
+    deltaChatPrimaryFg: theme.textPrimary, // todo rename this var
+    deltaChatPrimaryFgLight: theme.textSecondary, // todo rename this var
+    // Context Menu
+    contextMenuBg: theme.bgSecondary,
+    contextMenuBorder: undefinedGuard(
+      theme.bgSecondary, c => changeContrast(c, 0.1)
+    ),
+    contextMenuText: theme.textSecondary,
+    contextMenuSelected: theme.bgSecondary,
+    contextMenuSelectedBg: '#a4a6a9',
+    // Misc
+    avatarLabelColor: '#ffffff', // Only changable with theme.raw
+    bp3DialogHeaderBg: theme.bgSecondary,
+    bp3DialogBg: theme.bgPrimary,
+    bp3DialogCardBg: theme.bgSecondary,
+    bp3Heading: theme.textPrimary,
+    bp3ButtonText: theme.textPrimary,
+    bp3ButtonBg: theme.bgPrimary,
+    brokenMediaText: '#070c14',
+    brokenMediaBg: '#ffffff',
+    unreadCountBg: theme.accentColor,
+    unreadCountLabel: '#ffffff', // Only changable with theme.raw
+    contactListItemBg: '#62656a',
+    errorColor: '#f44336',
+    globalLinkColor: '#2090ea', // Only changable with theme.raw
+    globalBackground: theme.bgPrimary,
+    globalText: theme.textPrimary,
+    mapOverlayBg: theme.bgPrimary,
+    videoPlayBtnIcon: theme.accentColor,
+    videoPlayBtnBg: '#ffffff', // Only changable with theme.raw
+    scrollBarThumb: undefinedGuard(
+      theme.scrollbarTransparency, c => Color('black').alpha(c).rgb().string()
+    ),
+    scrollBarThumbHover: undefinedGuard(
+      theme.scrollbarTransparency, c => Color('black').alpha(c + 0.14).rgb().string()
+    )
```

Comment:
I don't know how much work this adds that if we want to define a new variable, we need to hardcode it at several places. Also it seems to me that we can't override those variables here from a theme?

Reply:
we need to hardcode it once for styled components and 3-4 times for scss

Reply:
Also for scss: we could just transfer the ("highlevel") theme vars and resolve them in scss, but some of them need additional transformation that must be done in js, because at runtime we don't have the fancy scss functions.

Reply:
@Simon-Laux what kind of additional transformation do they need?

Reply:
The actual theme gets generated here from the "highlevel" theme the user specified, after that when the "raw" object is set, its values replace the the generated values, so you still can have whole controll of it.